### PR TITLE
Update build

### PR DESCRIPTION
**Pull request by gschivley**
Updates to the pyproj.toml
- Back-end build system
- New maintainers
- Docs ULR
- Add "black" as an optional dependency for [dev]

Try adding a GitHub action to test building

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+
+    - name: Install Conda environment with Micromamba
+      uses: mamba-org/setup-micromamba@v1
+      with:
+        environment-file: environment.yml
+        cache-environment: true
```
Comment (lbonaldo) [inline .github/workflows/pytest.yml]:
```suggestion
      - name: Install Conda environment with Micromamba
        uses: mamba-org/setup-micromamba@v1
        with:
          environment-file: environment.yml
          cache-environment: true
```

**Review by lbonaldo — APPROVED** [review]
Thanks @gschivley. I added a minor change, but it looks good to me overall. I'm curious about the choice of `hatchling` over `setuptools`.

**Comment by gschivley** [comment]
@lbonaldo the switch to hatchling was as much to try it out. Python's [packaging guide](https://packaging.python.org/en/latest/tutorials/packaging-projects/#choosing-a-build-backend) now uses hatchling as their default so it seemed like a safe choice.

**Comment by gschivley** [comment]
The new build `build_test.yml` workflow [was successful on my fork](https://github.com/gschivley/EnergyPATHWAYS/actions/runs/8304480637/job/22730165350).

**Comment by lbonaldo** [comment]
> @lbonaldo the switch to hatchling was as much to try it out. Python's [packaging guide](https://packaging.python.org/en/latest/tutorials/packaging-projects/#choosing-a-build-backend) now uses hatchling as their default so it seemed like a safe choice.

Great! Thanks for updating it.

**Comment by lbonaldo** [comment]
> The new build `build_test.yml` workflow [was successful on my fork](https://github.com/gschivley/EnergyPATHWAYS/actions/runs/8304480637/job/22730165350).

I also tested it and all the jobs are completed.

**Pull request merged**